OMEdit/OMEditLIB/Simulation/SimulationOutputWidget.cpp

```diff
@@ -575,6 +575,73 @@ void SimulationOutputWidget::compileModel()
 }
 
 
+/*!
+ * \brief SimulationOutputWidget::runPostCompilation
+ * Runs the post compilation command after the compilation of the model.
+ */
+void SimulationOutputWidget::runPostCompilation()
+{
+  const QString postCompilationCommand = OptionsDialog::instance()->getSimulationPage()->getPostCompilationCommand();
+  if (postCompilationCommand.size())
+  {
+    mpCompilationProcess = new QProcess;
+    mpCompilationProcess->setWorkingDirectory(mSimulationOptions.getWorkingDirectory());
+    connect(mpCompilationProcess, SIGNAL(started()), SLOT(compilationProcessStarted()));
+    connect(mpCompilationProcess, SIGNAL(readyReadStandardOutput()), SLOT(readCompilationStandardOutput()));
+    connect(mpCompilationProcess, SIGNAL(readyReadStandardError()), SLOT(readCompilationStandardError()));
+  #if (QT_VERSION >= QT_VERSION_CHECK(5, 6, 0))
+    connect(mpCompilationProcess, SIGNAL(errorOccurred(QProcess::ProcessError)), SLOT(compilationProcessError(QProcess::ProcessError)));
+  #else
+    connect(mpCompilationProcess, SIGNAL(error(QProcess::ProcessError)), SLOT(compilationProcessError(QProcess::ProcessError)));
+  #endif
+    connect(mpCompilationProcess, SIGNAL(finished(int, QProcess::ExitStatus)), SLOT(postCompilationProcessFinished(int, QProcess::ExitStatus)));
+    writeCompilationOutput(QString("%1\n").arg(postCompilationCommand), Qt::black);
+    mpCompilationProcess->start(postCompilationCommand);
+  }
+  else
+  {
+    // no post-compilation step, run directly the simulation
+    if (!mSimulationOptions.getBuildOnly() && !mSimulationOptions.getLaunchAlgorithmicDebugger()) {
+      runSimulationExecutable();
+    }
+  }
+}
+
+/*!
+ * \brief SimulationOutputWidget::postCompilationProcessFinished
+ * Slot activated when mpCompilationProcess finished signal is raised.\n
+ * If the mpCompilationProcess finished normally then run the simulation executable.\n
+ * \param exitCode
+ * \param exitStatus
+ */
+void SimulationOutputWidget::postCompilationProcessFinished(int exitCode, QProcess::ExitStatus exitStatus)
+{
+  mIsCompilationProcessRunning = false;
+  QString exitCodeStr = tr("Post compilation process failed. Exited with code %1.").arg(Utilities::formatExitCode(exitCode));
+  if (exitStatus == QProcess::NormalExit && exitCode == 0) {
+    writeCompilationOutput(tr("Post compilation process finished successfully.")+"\n", Qt::blue);
```

Comment:
You can use `\n` directly after the dot. The localizations can handle escape characters.

Reply:
yes, but that avoids having to re-write translations

Reply:
ok